pymc/sampling/mcmc.py

```diff
@@ -293,8 +299,24 @@ def _sample_external_nuts(
                 "`idata_kwargs` are currently ignored by the nutpie sampler",
                 UserWarning,
             )
-
+        # gather observed and constant data as nutpie.sample() has no access to the PyMC model
+        coords, dims = coords_and_dims_for_inferencedata(model)
```

Comment:
This can be done after sampling (if sampling fails, there's no point in doing this work in advance). 

Please add a comment along the lines of `# Temporary work-around. Revert once https://github.com/pymc-devs/nutpie/issues/74 is fixed`

Reply:
It's a very cheap function call: https://github.com/pymc-devs/pymc/blob/6f4a04017e2d32d64fcb85a2390965c9aa73f6a4/pymc/backends/arviz.py#L103-L112

And sometimes this step catches issues with coordinate value types, that's why I recommended to put it before sampling (just like gathering constant and observed data)